python/lsst/ts/m2gui/controltab/tab_utility_view.py

```diff
@@ -371,6 +384,58 @@ def _create_group_displacements(self):
 
         return create_group_box("Displacement Sensors", layout)
 
+    def _update_power_system_status(self):
+        """Update the power system status."""
+
+        power_system_status = self.model.controller.power_system_status
+
+        # Power
+        power_motor = "On" if power_system_status["motor_power_is_on"] else "Off"
+        power_communication = (
+            "On" if power_system_status["communication_power_is_on"] else "Off"
+        )
+
+        self._power_inclinometer["power_on_motor"].setText(f"{power_motor}")
+        self._power_inclinometer["power_on_communication"].setText(
+            f"{power_communication}"
+        )
+
+        # State
+        state_motor = power_system_status["motor_power_state"]
+        state_communication = power_system_status["communication_power_state"]
+
+        self._power_inclinometer["power_system_state_motor"].setText(
+            f"{state_motor.name}"
+        )
+        self._power_inclinometer["power_system_state_communication"].setText(
+            f"{state_communication.name}"
+        )
+
+    def _set_signal_power_system(self, signal_power_system):
+        """Set the power system signal with callback function. This signal
+        reports the update of power system status.
+
+        Parameters
+        ----------
+        signal_power_system : `SignalPowerSystem`
+            Signal of the power system.
+        """
+        signal_power_system.is_state_updated.connect(
+            self._callback_update_power_system_status
+        )
+
+    @asyncSlot()
+    async def _callback_update_power_system_status(self, is_state_updated):
```

Comment:
This round of private functions can be simplified by passing data with signal. More in person

Reply:
This is to be consistent in the code base to improve the code maintainability. Thanks!